specs/0118-notification-center.md

```diff
@@ -0,0 +1,134 @@
+- Feature Name: Messaging Center
+- Created: 2024-03-12
+- Associated Issue: 
+
+# Summary
+
+Introduce a Messaging Center feature to make Pontoon the best tool to send communications to localizers directly, allowing sending of targeted in-app notifications and emails within Pontoon itself.
+
+# Motivation
+
+Currently, we can only send in-app notifications with limited targeting capabilities (selected locales within a single product). For any other communications targeting a subset of users we need to rely on engineers to extract email addresses and send emails to the targeted recipients. This feature would allow administrators of a Pontoon instance managing localization projects to send notifications or emails to targeted sets of recipients with more refined criteria such as locales, projects, roles, recency of activity, and level of activity.
+
+# Feature explanation
+
+## Messaging Center
+
+This feature has the ability to compose a message with a subject and body, send a message as an in-app notification and/or email, and target a set of recipients based on commonly used criteria.
+
+### Composing
+
+#### Message Editor
+
+- Top menu item: `Messaging`
+  - Item that appears in the top menu only for those with administrator priveleges
+  - Navigates to the `Messaging Center` page.
+- Title: `Messaging Center`
+- Header: `Message type`
+  - Selects the type of message to be sent
+  - Checkboxes: `Email`, `Notification` (default: both boxes deselected)
+  - Validation: Error if no checkbox selected - `You must select at least one message type.`
+- Header: `Message editor`
+  - Creates a subject and body for emails. Subject line will also be required for in-app notifications, which will have it combined with the body as the first line of the message separated by a newline.
+  - Field: `Subject` (text input)
+  - Field: `Body` (text input with formatting: markdown or html)
+  - Tab: `Preview` - shows a rendered version of the text in `Body` for checking formatting, similar to how the preview for GitHub comments work
+  - Validation: 
+    - Error if `Subject` field empty: `Your message must include a subject.`
+    - Error if `Body` field empty: `Your message must include a body.`
+
+#### Selecting Recipients with Filters
+
+- Header: `Recipients`
+  - Select who will receive a message based on a combination of the following filters.
+  - `Included User Roles` - enables User Role as a filter.
+    - Checkboxes: `Manager`, `Translator`, `Contributor` (multiselect, default: all boxes selected)
+    - Validation:
+      - Error if `Included User Roles` has no roles selected: `You must select at least one user role.`
+  - Subheader: `Filter by Locale`
```

Comment:
Default value is not specified. I assume it should be all Locales selected. And same for Projects below.